includes/namespace.php

```diff
@@ -48,6 +48,7 @@ function activate( $network_wide ) {
 
 	$story = new Story_Post_Type( new Experiments() );
 	$story->init();
+	$story->add_caps_to_roles();
```

Comment:
Is this activate callback run when the plugin is activated network wide and when a new site is created? Before this wasn't really a problem, now this activate HAS to run or you will not have access to the post type.

Reply:
So maybe call on `wp_insert_site`  action.
```
function create_new_site( $new_site ){
      switch_to_blog( $new_site );
      activate( false );
      restore_blog();
}
```

Reply:
From another plugin 

```
	// Activate plugin when new blog is added
	add_action( 'wpmu_new_blog', array( $this, 'activate_new_site' ) );
```

```
/**
	 * Fired when a new site is activated with a WPMU environment.
	 *
	 * @since    1.0.0
	 *
	 * @param    int $blog_id ID of the new blog.
	 */
	public function activate_new_site( $blog_id ) {

		if ( 1 !== did_action( 'wpmu_new_blog' ) ) {
			return;
		}

		switch_to_blog( $blog_id );
		self::single_activate();
		restore_current_blog();

	}
```

Reply:
No it isn't, see #2929 that you've created. Maybe a good opportunity to tackle that ticket next sprint?

Reply:
That is funny that I forgot I created that issue. Okay let's deal with it in that issue. I have assigned the ticket to myself.